lib/rucio/web/rest/flaskapi/v1/common.py

```diff
@@ -22,22 +22,59 @@
 from typing import TYPE_CHECKING
 
 import flask
+import typing
 from flask.views import MethodView
 from werkzeug.datastructures import Headers
 from werkzeug.exceptions import HTTPException
+from werkzeug.wrappers import Request, Response
 
 from rucio.api.authentication import validate_auth_token
 from rucio.common.exception import RucioException, CannotAuthenticate, UnsupportedRequestedContentType
 from rucio.common.schema import get_schema_value
 from rucio.common.utils import generate_uuid, render_json
 from rucio.core.vo import map_vo
+from rucio.common import config
+from configparser import NoOptionError, NoSectionError
+
 
 if TYPE_CHECKING:
     from typing import Optional, Union, Dict, Sequence, Tuple, Callable, Any, List
 
     HeadersType = Union[Headers, Dict[str, str], Sequence[Tuple[str, str]]]
 
 
+class CORSMiddleware(object):
+    """
+    WebUI 2.0 makes preflight requests to the API, which are not handled by the API.
+    This middleware intercepts the preflight OPTIONS requests and returns a 200 OK response.
+    """
+
+    def __init__(self, app: flask.Flask) -> typing.NoReturn:
+        self.app = app
+
+    def __call__(self, environ: typing.Dict, start_response: typing.Callable) -> typing.Union[Response, typing.Iterable[bytes]]:
+        request: Request = Request(environ)
+
+        if request.environ.get('REQUEST_METHOD') == 'OPTIONS':
+            try:
+                webui_urls = config.config_get_list('webui', 'urls')
```

Comment:
The `origin` will be the host though? So for atlas it wouldn't be `rucio-ui.cern.ch` but some random k8s hostname? I am just wondering, because this means we need to handle this list of webui urls in the config dynamically.

Reply:
As discussed with @rcarpa, the origin should be the public-facing URL the end-users will utilize. So, we would not have to maintain a list of dynamically created pods that would host the webui, ;)